perfkitbenchmarker/benchmarks/scimark2_benchmark.py

```diff
@@ -0,0 +1,288 @@
+  ]
+  for cmd in cmds:
+    stdout, _ = vm.RemoteCommand(cmd, should_log=True)
+    samples.extend(ParseResults(stdout))
+  return samples
+
+
+def Cleanup(benchmark_spec):
+  """Cleanup SciMark2 on the target vm.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  pass
+
+
+def ParseResults(results):
+  """Result parser for SciMark2.
+
+  Sample Results (C version):
+    **                                                              **
+    ** SciMark2 Numeric Benchmark, see http://math.nist.gov/scimark **
+    ** for details. (Results can be submitted to pozo@nist.gov)     **
+    **                                                              **
+    Using       2.00 seconds min time per kenel.
+    Composite Score:         1596.04
+    FFT             Mflops:  1568.64    (N=1024)
+    SOR             Mflops:  1039.98    (100 x 100)
+    MonteCarlo:     Mflops:   497.64
+    Sparse matmult  Mflops:  1974.39    (N=1000, nz=5000)
+    LU              Mflops:  2899.56    (M=100, N=100)
+
+  (Yes, "kenel" is part of the original output.)
+
+  Sample Results (Java version):
+
+    SciMark 2.0a
+
+    Composite Score: 1731.4467627163242
+    FFT (1024): 996.9938397943672
+    SOR (100x100):   1333.5328291027124
+    Monte Carlo : 724.5221517116782
+    Sparse matmult (N=1000, nz=5000): 1488.18620413327
+    LU (100x100): 4113.998788839592
+
+    java.vendor: Oracle Corporation
+    java.version: 1.7.0_75
+    os.arch: amd64
+    os.name: Linux
+    os.version: 3.16.0-25-generic
+
+  Args:
+    results: SciMark2 result.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  RESULT_START_REGEX = re.compile(r'^;;; \s+ (.*)', re.X | re.M)
+
+  SCORE_REGEX = re.compile(r'''
+    ^ (Composite \s+ Score) : \s+ (\d+ \. \d+)
+  ''', re.X | re.M)
+  RESULT_REGEX_C = re.compile(r'''
+    ^
+    ( .+? ) \s+  #1: Test name
+    Mflops: \s+
+    ( \d+ \. \d+ )  #2: Test score
+    ( \s+ \( .+? \) )?  #3: Optional test details
+  ''', re.X | re.M)
+  RESULT_REGEX_JAVA = re.compile(r'''
+    ^
+    ( .+? )  #1: Test name
+    : \s+
+    ( \d+ \. \d+ )  #2: Test score
+  ''', re.X | re.M)
+  PLATFORM_REGEX = re.compile(r'''
+    ^
+    ( \w+ \. \w+ )  #1: Property name
+    : \s+
+    ( .* )  #2: Property value
+  ''', re.X | re.M)
+
+  def FindBenchStart(results, start_index=0):
+    m = RESULT_START_REGEX.search(results, start_index)
+    if m is None:
+      return -1, 'Unknown'
+    return m.start(), m.group(1)
+
+  def ExtractPlatform(result, bench_version):
+    metadata = {}
+    meta_start = None
+    if bench_version.startswith('C'):
+      pass
+    elif bench_version.startswith('Java'):
+      for m in PLATFORM_REGEX.finditer(result):
+        if meta_start is None:
+          meta_start = m.start()
+        metadata[m.group(1)] = m.group(2)
+    return metadata, meta_start
+
+  def ExtractScore(result, bench_version):
+    m = SCORE_REGEX.search(result)
```

Comment:
This will raise a fairly uninformative error if `SCORE_REGEX` does not match `result`.